README.md

```diff
@@ -1,2 +1,28 @@
 # dune-adapter
 **experimental** preCICE-adapter for DUNE, a modular toolbox for solving partial differential equations
+
+A adapter module enabeling multiphysics simulation inside DUNE with the help of preCICE [1][2].
+
+The dune-adapter module depends on the DUNE core modules and
+on dune-elastodynamics providing the necessary tools for structural simulation.
+
+The dune-elastodynamics module can be found here:
+https://github.com/maxfirmbach/dune-elastodynamics
```

Comment:
Cloning another repository is somewhat tedious. Would it be possible to include the files here? Or add a `git submodule`?

Reply:
The `foamgrid` module is a dependency by the elastodynamics module and non-default module (at least not available via apt). This should be documented somewhere.

Reply:
Building of the elastodynamics module fails with the first error message being:

```bash
In file included from dune-adapter/dune-elastodynamics/dune/elastodynamics/timesteppers/newmark.hh:10,
                 from dune-adapter/dune-elastodynamics/src/Example03.cc:36:
dune-adapter/dune-elastodynamics/dune/elastodynamics/preconditioners/distributedscalarproduct.hh: In instantiation of ‘class DistributedScalarProduct<Dune::GridView<Dune::UGGridLeafGridViewTraits<const Dune::UGGrid<2> > >, Dune::BlockVector<Dune::FieldVector<double, 2> > >’:
dune-adapter/dune-elastodynamics/dune/elastodynamics/timesteppers/newmark.hh:55:24:   required from ‘void Dune::Newmark<GridView, MatrixType, VectorType>::initialize(VectorType&, VectorType) [with GridView = Dune::GridView<Dune::UGGridLeafGridViewTraits<const Dune::UGGrid<2> > >; MatrixType = Dune::BCRSMatrix<Dune::FieldMatrix<double, 2, 2> >; VectorType = Dune::BlockVector<Dune::FieldVector<double, 2> >]’
dune-adapter/dune-elastodynamics/src/Example03.cc:110:46:   required from here
dune-adapter/dune-elastodynamics/dune/elastodynamics/preconditioners/distributedscalarproduct.hh:32:22: error: ‘typename Dune::ScalarProduct<Vector>::field_type DistributedScalarProduct<GridView, Vector>::dot(const Vector&, const Vector&) const [with GridView = Dune::GridView<Dune::UGGridLeafGridViewTraits<const Dune::UGGrid<2> > >; Vector = Dune::BlockVector<Dune::FieldVector<double, 2> >; typename Dune::ScalarProduct<Vector>::field_type = double]’ marked ‘override’, but does not override
```

Reply:
So I have one Folder, where all the dune-modules are in (I cloned all of them).

Reply:
Do you have a requirement on a specific version? Running `dunecontrol` works in the first place, but it is a compiler error as you can see.

Reply:
I use version 2.7 ... Two students also installed it and for them everything worked fine.

Reply:
Hm I installed version `2.6`. Do I need `dune-python`?

Reply:
I haven't installed `dune-python` ... I guess it wont work with version `2.6`, sorry, I also need to add to use at least `2.7`.

Reply:
You can add the required DUNE module versions to the `dune.module` file. [DuMuX is doing it that way as well.](https://git.iws.uni-stuttgart.de/dumux-repositories/dumux/-/blob/master/dune.module)

Reply:
I cloned them one by one to be honest.